js/client.js

```diff
@@ -2741,15 +2741,27 @@ function toId() {
 		initialize: function (data) {
 			var buf = '';
 			buf = '<p>Your replay has been uploaded! It\'s available at:</p>';
-			buf += '<p><a href="https://' + Config.routes.replays + '/' + data.id + '" target="_blank" class="no-panel-intercept">https://' + Config.routes.replays + '/' + data.id + '</a></p>';
-			buf += '<p><button class="autofocus" name="close">Close</button></p>';
+			buf += '<p> <a id="replayLink" href="https://' + Config.routes.replays + '/' + data.id + '" target="_blank" class="no-panel-intercept">https://' + Config.routes.replays + '/' + data.id + '</a></p>';
+			buf += '<p><button class="autofocus" name="close">Close</button> <button name="copyReplayLink">Copy</button><p>';
 			this.$el.html(buf).css('max-width', 620);
 		},
 		clickClose: function () {
 			this.close();
 		},
 		submit: function (i) {
 			this.close();
+		},
+		copyReplayLink: function () {
+			var copyText = document.querySelector("#replayLink");
+			var dummyReplayLink = document.createElement("input");
+			// This is a hack. You can only "select" an input field. The trick is to create a short lived input element and destroy it after a copy.
+			dummyReplayLink.id = "dummyReplayLink";
+			dummyReplayLink.value = copyText.href;
+			dummyReplayLink.style = "absolute; left: -1000px; top: -1000px";
```

Comment:
```suggestion
			dummyReplayLink.style = "position: absolute; left: -1000px; top: -1000px";
```

Does this actually support being set to a string, though?

Reply:
> Does this actually support being set to a string, though?

Yeah it does. See [here](https://gomakethings.com/two-ways-to-set-an-elements-css-with-vanilla-javascript/). You can provide a semicolon delimited list of attributes `.style`.

Reply:
Good to know!

(You forgot the `position:` part.)

Reply:
Also your link doesn't actually say it's possible to provide a semicolon-delimited list like that.

Reply:
MDN says you can't:

https://developer.mozilla.org/en-US/docs/Web/API/ElementCSSInlineStyle/style

> Styles should not be set by assigning a string directly to the `style` property

I presume you probably want to use `setAttribute`, or by setting individual styles.

Reply:
Okay got it. Thanks @Zarel. I have fixed it now.